File: backend/src/main/java/com/NBE3_4_2_Team4/global/security/oauth2/logoutService/OAuth2LogoutService.java
//package com.NBE3_4_2_Team4.global.security.oauth2.logoutService;
//
//
//import com.NBE3_4_2_Team4.domain.member.member.entity.Member;
//import org.springframework.beans.factory.annotation.Value;
//
//public abstract class OAuth2LogoutService {
//    @Value("${custom.domain.backend}")
//    String backendDomain;
//
//    public static final String LOGOUT_COMPLETE_URL = "/api/logout/complete";
//
//    public abstract Member.OAuth2Provider getOAuth2Provider();
//
//    public String getLogoutRedirectUrl() {
//        return backendDomain + LOGOUT_COMPLETE_URL;
//    }
//
//    public String getLogoutUrl(){
//        return getLogoutRedirectUrl();
//    };
//}

File: backend/src/main/java/com/NBE3_4_2_Team4/global/security/oauth2/logoutService/OAuth2LogoutService.kt
package com.NBE3_4_2_Team4.global.security.oauth2.logoutService

import com.NBE3_4_2_Team4.domain.member.member.entity.Member
import org.springframework.beans.factory.annotation.Value

abstract class OAuth2LogoutService {
    @Value("\${custom.domain.backend}")
    lateinit var backendDomain: String

    companion object {
        const val LOGOUT_COMPLETE_URL: String = "/api/logout/complete"
    }

    abstract fun getOAuth2Provider(): Member.OAuth2Provider

    fun getLogoutRedirectUrl(): String {
        return backendDomain + LOGOUT_COMPLETE_URL
    }

    open fun getLogoutUrl(): String {
        return getLogoutRedirectUrl()
    };
}